BrowserKit/Sources/Redux/Store.swift

```diff
@@ -68,15 +70,31 @@ public class Store<State: StateType>: DefaultDispatchStore {
     }
 
     public func dispatch(_ action: Action) {
-        guard Thread.isMainThread && !actionRunning else {
-            DispatchQueue.main.async { [weak self] in self?.dispatch(action) }
-            return
-        }
-
         logger.log("Dispatched action: \(action.displayString())", level: .info, category: .redux)
 
-        actionRunning = true; defer { actionRunning = false }
+        if Thread.isMainThread {
```

Comment:
Very minor nit: this might possibly be more readable as a guard, e.g.

```
guard Thread.isMainThread else {
    DispatchQueue.main.async { [weak self] in self?.dispatch(action) }
    return
}

actionQueue.append(action)
processQueuedActions()
```

NAB though. Other than the comment below LGTM so far.